rlog/2024-10-22-gsub-idontwant-tests.mdx

```diff
@@ -0,0 +1,110 @@
+---
+title: 'IDONTWANT Message Potential: Performance Insights and Connected Findings'
+date: 2024-10-22 12:00:00
+authors: farooq
+published: true
+slug: gsub-idontwant-tests
+categories: research
+discuss: https://forum.vac.dev/t/idontwant-message-potential-performance-insights-and-connected-findings/374
+
+
+toc_min_heading_level: 2
+toc_max_heading_level: 5
+---
+
+This post provides quick insights into the IDONTWANT message performance and highlights minor tweaks that can further contribute to performance gains.
+<!--truncate-->
+
+## Overview
+
+[IDONTWANT](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.2.md?plain=1#L52) messages are introduced to curtail redundant transmissions without compromising resilience. 
+Cutting down on duplicates can potentially render two significant advantages
+
+1. Minimize bandwidth requirements 
+2. Minimize message dissemination time
+
+For IDONTWANTs to be effective, they must be received and processed by the sender before it starts relaying the message. 
+[Duplicates investigation](https://ethresear.ch/t/number-duplicate-messages-in-ethereums-gossipsub-network/19921#arrival-time-of-duplicates-9) reveals that 
+the average time difference between the first message arrival and the first duplicate arrival is higher than the average round trip time in Ethereum's GossipSub network. 
+
+This allows for timely IDONTWANT reception and canceling of many duplicate transmissions, 
+showing a potential for a significant drop in bandwidth utilization. 
+On the other hand, lowering message dissemination time is only possible by minimizing queuing delays at busy peers. 
+
+## Experiments
+
+We conducted a series of experiments with different arrangements (changing heartbeat_interval and message size) 
+to precisely identify the impact of IDONTWANT messages on bandwidth utilization and message dissemination time. 
+
+The experiments are performed on nim-libp2p using the [shadow simulator](https://github.com/vacp2p/dst-gossipsub-test-node/pull/4). 
+The peer bandwidth and link latency are uniformly set between 50-150 Mbps and 40-130 milliseconds in five stages. 

```

Comment:
What it means that the means that a peer bandwidth is set between 50-150 Mbps? To me it sounds that is the minimum, but is that minimum over the entire simulation? Or how it works? Could you also point so some documentation regarding this?

Reply:
> What it means that the means that a peer bandwidth is set between 50-150 Mbps? To me it sounds that is the minimum, but is that minimum over the entire simulation? Or how it works? Could you also point so some documentation regarding this?

every peer have bandwidth between 50-150. we have five stages (50, 75, 100,...). details here in readme https://github.com/vacp2p/dst-gossipsub-test-node/pull/4